src/chapters/how-to-write-code/input.rst

```diff
@@ -0,0 +1,176 @@
+together to make this process work. For now, here is a brief overview.
+
+Load the Module
+^^^^^^^^^^^^^^^^
+
+In line 1 ``require(`readline-sync`)`` pulls in all the functions that allow
+us to get data from the user.
+
+``input =`` assigns all of these functions to ``input``.
+
+The meaning of ``const`` will be disucssed in the next chapter.
+
+How to Prompt the User
+^^^^^^^^^^^^^^^^^^^^^^^
+
+To display a prompt and wait for a response, we use the following syntax:
+``input.question("Question text... ");``.
+
+When JavaScript evaluates the expression, it follows the instructions:
+
+#. Display ``Question text`` on the screen.
+#. Wait for the user to respond.
+
+For our greeting program, we would code ``input.question("Enter your name:
+");``. The user enters a name and presses the Return or Enter key. When
+this happens, any text entered is collected by the input function.
+
+.. admonition:: Try It
+
+   Let's play around with the ``input`` statement. Open the repl.it link below
+   and click the "Run" button.
+
+   .. replit:: js
+      :slug: InputExamples01
+      :linenos:
+
+      const input = require(`readline-sync`);
+
+      input.question("Enter your name: ");
+
+Note that after entering a name, the program does not actually DO anything with
+the information. If we want to print the data as part of a message, we need to
+put it inside a ``console.log`` statement. ``input.question()`` can be used
+just like any other sample of text.
+
+Change line 3 to
+``console.log("Hello, " + input.question("Enter your name: "));``, then run the
+code several times, trying different responses to the input prompt. The program
+collects the user's information first, and then prints the output.
+
+Storing the Data for Later
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As written, our code prompts the user for information just before that data
+gets used. What if we need to collect input but use it at a later time?
+
+JavaScript (and all other programming languages) allow us to store data inside
+a container called a *variable*. Do not worry about the nitty-gritty technical
+details here. In the next chapter we dive deep into the concepts behind data
+and variables.
+
+For now, let's assign a container called ``name`` to store the name entered by
+the user. Modify your code as follows:
+
+.. sourcecode:: js
+   :linenos:
+
+   const input = require(`readline-sync`);
+
+   let name = input.question("Enter your name: ");
+
+   console.log("Hello, " + name);
+
+Run the new code several times, and try different responses to the input
+prompt. Note how the code stores the data inside ``name``, and then recalls
+that information by adding ``name`` to the ``console.log`` statement in line 5.
+
+Unlike ``console.log("Hello, " + input.question("Enter your name: "));``, by
+storing the user's name inside ``name``, we gain the ability to hold onto the
+data and use it when and where we see fit.
```

Comment:
This sentence's length obscures its meaning. I would break it down.

Reply:
Fixed.